duckduckgo/macos-browser
# Makes the error, location and ip fields selectable in the NetP status…

Task/Issue URL: https://app.asana.com/0/1203137811378537/1204613002619098/f

## Description:

Makes a few fields in Network Protection's status view selectable.

The fields are:
- Error message field
- Server location field
- IP field

**Known limitations:** This change is limited to macOS 12+ and iOS 15+, as there wasn't a simple way to implement this functionality for older OSes.

## Testing:

### Test 1: Error Message

1. Don't start NetP just yet
2. Make sure this is enabled: Debug Menu > Network Protection > Simulate Failure > "Enable NetP > Controller failure"
3. Connect NetP, an error should come up in the status view
4. Select the text, copy and paste elsewhere

You should be able to make that work.

### Test 2: Location and IP fields

1. Start NetP.
2. Copy the server location by selecting the text
3. Paste it elsewhere
4. Copy the IP from NetP's status view by selecting the text
5. Past it elsewhere

You should be able to copy those fields and paste elsewhere.

---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)
